src/test/java/com/google/cloud/genomics/utils/grpc/VariantUtilsTest.java

```diff
@@ -27,13 +30,347 @@
+            Variant.newBuilder(multiTemplate).addCalls(haploidRef).addCalls(homRef).build()));
+  }
+
+  @Test
+  public void testIsFiltered() {
+    Variant variant =
+        Variant.newBuilder()
+            .setId("Var1")
+            .addFilter(VariantUtils.PASSES_FILTERS)
+            .setReferenceBases("A")
+            .addAlternateBases("G")
+            .addAlternateBases("C")
+            .addAlternateBases("T")
+            .build();
+    assertFalse(VariantUtils.IS_FILTERED.apply(variant));
+
+    variant =
+        Variant.newBuilder()
+            .setId("Var2")
+            .addFilter(VariantUtils.MISSING_VALUE)
+            .setReferenceBases("A")
+            .addAlternateBases("G")
+            .addAlternateBases("C")
+            .addAlternateBases("T")
+            .build();
+    assertFalse(VariantUtils.IS_FILTERED.apply(variant));
+
+    variant =
+        Variant.newBuilder()
+            .setId("Var3")
+            .addFilter("BAD_SB_SCORE")
+            .setReferenceBases("A")
+            .addAlternateBases("G")
+            .addAlternateBases("C")
+            .addAlternateBases("T")
+            .build();
+    assertTrue(VariantUtils.IS_FILTERED.apply(variant));
+
+    variant =
+        Variant.newBuilder()
+            .setId("Var4")
+            .addFilter("LowQual")
+            .addFilter("!ReallyBadHaplotype_score")
+            .setReferenceBases("A")
+            .addAlternateBases("G")
+            .addAlternateBases("C")
+            .addAlternateBases("T")
+            .build();
+    assertTrue(VariantUtils.IS_FILTERED.apply(variant));
+
+    // Test that variants without any filters defined are passing.
+    variant = Variant.newBuilder().setReferenceBases("A").addAlternateBases("G").build();
+    assertFalse(VariantUtils.IS_FILTERED.apply(variant));
+  }
+
+  @Test
+  public void testIsBiallelic() {
+    // no alternates
+    assertFalse(VariantUtils.IS_BIALLELIC.apply(Variant.newBuilder().setReferenceBases("A").build()));
+    // multiple alternates
+    assertFalse(VariantUtils.IS_BIALLELIC.apply(
+            Variant.newBuilder()
+                .setReferenceBases("A")
+                .addAlternateBases("G")
+                .addAlternateBases("T")
+                .build()));
+    // deletion
+    assertTrue(VariantUtils.IS_BIALLELIC.apply(
+            Variant.newBuilder().setReferenceBases("AC").addAlternateBases("A").build()));
+    // insertion
+    assertTrue(VariantUtils.IS_BIALLELIC.apply(
+            Variant.newBuilder().setReferenceBases("A").addAlternateBases("AT").build()));
+    // success
+    assertTrue(VariantUtils.IS_BIALLELIC.apply(
+            Variant.newBuilder().setReferenceBases("A").addAlternateBases("G").build()));
+  }
+
+  @Test
+  public void testIsGVCF() {
+    Variant variant = Variant.newBuilder()
+        .setReferenceBases("A")
+        .addAlternateBases("G")
+        .build();
+    assertFalse(VariantUtils.IS_GVCF.apply(variant));
```

Comment:
I'm sorry but this would be a valid gVCF variant.  Why is it not considered valid?  I think the Predicate criteria above needs to be updated.